internal/weldr/api.go

```diff
@@ -2357,8 +2357,13 @@ func (api *API) composeHandler(writer http.ResponseWriter, request *http.Request
 	if testMode == "1" || testMode == "2" {
 		// Fake a parent commit for test requests
 		cr.OSTree.Parent = "02604b2da6e954bd34b8b82a835e5a77d2b60ffa"
-	} else {
-		ref, checksum, err := ostree.ResolveParams(cr.OSTree, imageType.OSTreeRef())
+	} else if imageType.OSTreeRef() != "" {
+		// If the image type has a default ostree ref, assume this is an OSTree image
```

Comment:
@achilleas-k is this a reasonable assumption to make? I found out this bit got executed every time, and the way the default ref worked is that ResolveParams actually allowed an empty ref to be returned (if it isn't an ostree image).

Reply:
Yes, all implementations of that method look like this:

```go
func (t *imageType) OSTreeRef() string {
	d := t.arch.distro
	if t.rpmOstree {
		return fmt.Sprintf(d.ostreeRefTmpl, t.Arch().Name())
	}
	return ""
}
```

and that's generally how we identify ostree-based images when we need to.

Reply:
perfect, I'll use this as much as possible then to be consistent